byterun/alloc.c

```diff
@@ -83,6 +83,84 @@ CAMLexport value caml_alloc_small_with_my_or_given_profinfo (mlsize_t wosize,
   }
 }
 
+static inline value alloc_fixed(mlsize_t wosize, tag_t tag, value* vals)
+{
+  value result;
+  mlsize_t i;
+#undef Setup_for_gc
+#define Setup_for_gc CAMLparamN(vals, wosize)
+#undef Restore_after_gc
+#define Restore_after_gc CAMLdrop
+  Alloc_small (result, wosize, tag);
+#undef Setup_for_gc
+#define Setup_for_gc
+#undef Restore_after_gc
+#define Restore_after_gc
+  for (i = 0; i < wosize; i++) {
+    Field(result, i) = vals[i];
```

Comment:
I haven't deleted `Field`! There are many, many uses of `Field` in the ocaml runtime, and this patch does not remove them all (and even adds the above one). Only code compiled with `-DCAML_API_VERSION=405` cannot see `Field`. See the implementation on the multicore branch for a `Field`-less version.